app/views/add.scala.html

```diff
@@ -14,74 +14,61 @@
  * limitations under the License.
  *@
 
-@import models.User
+@import config.SiProtectedUserConfig
+@import play.api.data.Form
+@import play.api.i18n.Messages
+@import play.api.mvc.Request
 @import uk.gov.hmrc.govukfrontend.views.html.components._
 @import uk.gov.hmrc.govukfrontend.views.html.components.implicits._
+@import uk.gov.hmrc.hmrcfrontend.views.Implicits.RichErrorSummary
+@import views.html.Layout
+@import views.html.fragments.Heading1
 
 @this(
         formTemplate: FormWithCSRF,
+        h1: Heading1,
         input: GovukInput,
-        mainTemplate: main_template,
-        govukTable: GovukTable
+        govukErrorSummary: GovukErrorSummary,
+        govukSelect: GovukSelect,
+        govukButton: GovukButton,
+        mainLayout: Layout
 )
 
-@(form: Form[_], allowlisted: List[User], notice: Option[String])(implicit request: Request[_], messages: Messages)
+@(form: Form[Entry], siProtectedUserConfig: SiProtectedUserConfig)(implicit request: Request[_], messages: Messages)
 
-@mainTemplate(title = Messages("home.page.title")) {
-    <h1 class="govuk-heading-l">@Messages("page.header")</h1>
-
-    @formTemplate(action = controllers.routes.SiProtectedUserController.submit, args = Symbol("autocomplete") -> "off", Symbol("novalidate") -> "") {
-
-        <fieldset class="form-field-group govuk-fieldset">
-            <div class="govuk-form-group">
-            @input(Input(label = Label(content = Text(Messages("page.scp")))).withFormFieldWithErrorAsHtml(form("name")))
-            </div>
-
-
-            <div class="govuk-form-group">
-            @input(Input(label = Label(content = Text(Messages("page.org")))).withFormFieldWithErrorAsHtml(form("org")))
-            </div>
-
-
-            <div class="govuk-form-group">
-            @input(Input(label = Label(content = Text(Messages("page.requester.email")))).withFormFieldWithErrorAsHtml(form("requester_email")))
-            </div>
-
-            <div class="govuk-form-group">
-
-                <button class="govuk-button" type="submit">@Messages("page.add")</button>
-                <a class="govuk-button govuk-button--secondary" type="button" href="@routes.SiProtectedUserController.reload">
-                @Messages("clear.button")
-                </a>
-                <a class="govuk-button govuk-button--secondary" type="button" href="@routes.SiProtectedUserController.homepage">
-                @Messages("delete.complete.button")
-                </a>
-            </div>
-        </fieldset>
+@mainLayout(
+    pageTitle = Messages("add.page.title"),
+    backLink = Some(BackLink(href = controllers.routes.SiProtectedUserController.homepage.url))
```

Comment:
This is very specific. Will it always be the case that users enter this page from the home page?

Reply:
as far as I can tell its the only way to get there, is there a less specific way to do back links?